videostore/indexer/indexer.go

```diff
@@ -0,0 +1,540 @@
+// Package indexer manages metadata for video files stored on disk.
+package indexer
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	// Blank importing SQLite3 driver for its side-effects.
+	// This registers the "sqlite3" driver with the database/sql package.
+	_ "github.com/mattn/go-sqlite3"
+	vsutils "github.com/viam-modules/video-store/videostore/utils"
+	"go.viam.com/rdk/logging"
+)
+
+const (
+	dbFileName             = "index.sqlite.db"
+	dbFileMode             = 0o750
+	segmentsTableName      = "segments"
+	videoFileSuffix        = ".mp4"
+	indexerRefreshInterval = 10 * time.Second
```

Comment:
@nicksanford @hexbabe What do we think about bumping the polling time down to 1 second?

At the current 10 second interval, we will have a 1/3 chance of inaccurately reporting the most recent segment.